crates/api_crud/src/post/create.rs

```diff
@@ -130,6 +131,12 @@ pub async fn create_post(
     }
   };
 
+  if let Some(scheduled_time) = data.scheduled_time {
```

Comment:
Check out the `check_expire_time` function, for how to check and convert API unix times into `Option<Datetime<Utc>>` . You'll also need to use that (or a similar one) for updating the post.